Game info

AmberMLiu opened the pull request:
Add select game use case and game info entity.

AmberMLiu committed:
Preliminary implementation of the GameInfo entity, including the Blackjack subclass.

AmberMLiu committed:
Refactor BlackjackGameInfo to use instance variables.

AmberMLiu committed:
Refactor GameLibraryView to work with a list of availble games and slightly reformated layout. Add placeholder interaction with game library buttons.

AmberMLiu committed:
Minor changes

AmberMLiu committed:
Add associated files for the select game use case.

AmberMLiu committed:
Implement the Game Setup Interface and some button functionality.

AmberMLiu committed:
Small changes (e.g. checkstyle issues).

AmberMLiu committed:
Merge branch 'main' into game-info

EstelleWang-UofT commented on src/main/java/interface_adapter/game_setup/GameSetupController.java:
shouldn't the controller implement the input boundary?

EstelleWang-UofT replied:
No actually, I think the controller shouldn't be an implementation of any interface.

EstelleWang-UofT reviewed (changes requested):
Minor changes should be made. Please see my comment on GameSetupController.java .
I am also worried about the TODOs you wrote. Maybe finish them before merging into main?
If the TODOs are assigned to another person, they can also make changes to this Game-info big part of the code.

AmberMLiu committed:
Fix GameLibraryController to adhere to CA. Complete 1 TODO.

AmberMLiu committed:
Add game description to game setup use case.

AmberMLiu commented:
Fixed the GameSetupController problem, and implemented one TODO. Other TODOs are not related to this feature, I placed them there as a place holder for other people.
